skdownscale/pointwise_models/gard.py

```diff
@@ -121,11 +149,32 @@ def _predict_one_step(self, lr_model, X):
         x = np.asarray(self.kdtree_.data)[inds]
         y = self.y_[inds]
 
-        # train linear regression model
-        lr_model.fit(x, y)
+        # figure out if there's a threshold of interest
+        if self.thresh is not None:
+            exceed_ind = y > self.thresh
+        else:
+            exceed_ind = np.array([True] * len(y))
```

Comment:
```suggestion
            exceed_ind = np.ones(len(y), dtype=bool)
```

I did a quick test here and I think it is much more performant (and direct) to construct the array this way.

```ipython
In [9]: y = np.arange(1000

In [10]: %timeit exceed_ind = np.array([True] * len(y))
36 µs ± 209 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)

In [11]: %timeit exceed_ind = np.ones(len(y), dtype=bool)
1.06 µs ± 3.06 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```